jbehave-support-core/src/main/java/org/jbehavesupport/core/ssh/SshHandler.java

```diff
@@ -216,4 +274,79 @@ protected String getSearchColumnName(ExamplesTable searchData) {
             throw new IllegalArgumentException("No search column found in example table");
         }
     }
+
+    private void storeInExtension(final MultiKey key, String logContent) {
+        if (serverLogXmlReporterExtension != null) {
+            serverLogXmlReporterExtension.registerLogContent(key, logContent);
+        } else {
+            log.warn("ServerLogXmlReportExtension is not registered, no logs will be present in report.");
+        }
+    }
+
+    private void storeTemplatesLogs(Map<String, List<SshTemplate>> sshTemplates) {
+        sshTemplates.entrySet().stream().forEach(entry ->
```

Comment:
Personally I would be inclined to split into multiple methods to improve readability, but just a suggestion 🙂

Reply:
To be honest I don't see what should be splitted :(